pkg/apis/core/v1/defaults.go

```diff
@@ -318,6 +318,24 @@ func SetDefaults_HTTPGetAction(obj *v1.HTTPGetAction) {
 		obj.Scheme = v1.URISchemeHTTP
 	}
 }
+
+// SetDefaults_Namespace adds a default label for all namespaces
+func SetDefaults_Namespace(obj *v1.Namespace) {
+	// TODO, remove the feature gate in 1.22
+	// we cant SetDefaults for nameless namespaces (generateName),
+	// we don't need to overwrite the pre-existing labelMetadataName
+	// This code needs to be kept in sync with the implementation that exists
+	// in Namespace Canonicalize strategy (pkg/registry/core/namespace)
+	if len(obj.Name) > 0 && obj.Labels[v1.LabelMetadataName] != obj.Name {
+		if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
```

Comment:
I'd put this as the outer if.

```
if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
    if len(obj.Name) > 0 {
        if obj.Labels == nil {
            obj.Labels = map[string]string{}
        }
        obj.Labels[v1.LabelMetadataName] = obj.Name
    }
}
```

Reply:
I was trying to avoid repeatedly calling the feature gate enablement check when listing all namespaces if a local check would show no change was needed

Reply:
putting it on the outside makes logical sense - will make that change , unless @liggitt  feels strongly that maybe this is a performance issue ?

Reply:
fixed it - @liggitt if any major objection lmk , will revert